certifier_service/certlib/certlib_policy.go

```diff
@@ -0,0 +1,440 @@
+	"crypto/rsa"
+	"crypto/sha256"
+	"fmt"
+	"google.golang.org/protobuf/proto"
+	"io/ioutil"
+	"os"
+
+	certprotos "github.com/vmware-research/certifier-framework-for-confidential-computing/certifier_service/certprotos"
+)
+
+func PolicyStoreNumEntries(ps *certprotos.PolicyStoreMessage) int {
+	return len(ps.Entries)
+}
+
+func PrintPolicyStoreEntry(e *certprotos.PolicyStoreEntry) {
+	fmt.Printf("Tag: %s, Type: %s, Value: ", e.GetTag(), e.GetType())
+	PrintBytes(e.GetValue())
+	fmt.Printf("\n")
+}
+
+func NewPolicyStoreEntry(tag string, valueType string, value []byte) *certprotos.PolicyStoreEntry {
+	e := new(certprotos.PolicyStoreEntry)
+	e.Tag = &tag
+	e.Type = &valueType
+	e.Value = value
+	return e
+}
+
+func NewPolicyStore(maxEnts int) *certprotos.PolicyStoreMessage {
+	ps := new(certprotos.PolicyStoreMessage)
+	me := int32(maxEnts)
+	ps.MaxEnts = &me
+	return ps
+}
+
+func PrintPolicyStore(ps *certprotos.PolicyStoreMessage) {
+	fmt.Printf("Maximum Entries: %d, current entries: %d\n", int(ps.GetMaxEnts()), len(ps.Entries))
+	for i := 0; i < len(ps.Entries); i++ {
+		fmt.Printf("   ")
+		PrintPolicyStoreEntry(ps.Entries[i])
+	}
+}
+
+func FindPolicyStoreEntry(ps *certprotos.PolicyStoreMessage, tag string, valueType string) int {
+	for i := 0; i < len(ps.Entries); i++ {
+		if ps.Entries[i].GetTag() == tag && ps.Entries[i].GetType() == valueType {
+			return i
+		}
+	}
+	return -1
+}
+
+func InsertOrUpdatePolicyStoreEntry(ps *certprotos.PolicyStoreMessage, tag string, valueType string, value []byte) bool {
+	ent := FindPolicyStoreEntry(ps, tag, valueType)
+	if ent >= 0 {
+		ps.Entries[ent].Value = value
+		return true
+	}
+	if len(ps.Entries) >= int(ps.GetMaxEnts()) {
+		return false
+	}
+	pse := NewPolicyStoreEntry(tag, valueType, value)
+	ps.Entries = append(ps.Entries, pse)
+	return true
+}
+
+func PolicyStoreDeleteEntry(ent int) {
+}
+
+func ConstructKeyForProtect(keyName string, keyType string) *certprotos.KeyMessage {
+
+	// only alg for now is "aes-256-cbc-hmac-sha256"
+	if keyType != "aes-256-cbc-hmac-sha256" {
+		fmt.Printf("ConstructKeyForProtect: only aes-256-cbc-hmac-sha256 is supported\n")
+		return nil
+	}
+
+	nakedKey := make([]byte, 64)
+	_, err := rand.Read(nakedKey)
+	if err != nil {
+		fmt.Printf("ConstructKeyForProtect: Can't construct iv\n")
+		return nil
+	}
+
+	k := new(certprotos.KeyMessage)
+
+	tn := TimePointNow()
+	if tn == nil {
+		fmt.Printf("ConstructKeyForProtect: Can't get TimeNow\n")
+		return nil
+	}
+
+	tf := TimePointPlus(tn, 365*86400)
+	if tf == nil {
+		fmt.Printf("ConstructKeyForProtect: Can't get TimeAfter\n")
+		return nil
+	}
+
+	nb := TimePointToString(tn)
+	na := TimePointToString(tf)
+
+	k.KeyName = &keyName
+	k.KeyType = &keyType
+	format := "vse-key"
+	k.KeyFormat = &format
+	k.SecretKeyBits = nakedKey
+	k.NotBefore = &nb
+	k.NotAfter = &na
+	return k
+}
+
+func ProtectBlob(enclaveType string, k *certprotos.KeyMessage, buffer []byte) []byte {
+
+	if k.GetKeyType() != "aes-256-cbc-hmac-sha256" {
```

Comment:
I found 3 instances of the hard-coded string `"aes-256-cbc-hmac-sha256"` in this file, alone.

I don't quite understand why we are only supporting just this one algorithm.  In any case, is it possible in Go to define a static string constant in this file? If yes, it will be better to create one named, say `Integrity_method_aes_256_cbc_hmac_sha256` (same as the one defined in `src/certifier_algorithms.cc`) and use that in all places in this Go-file.

And add some comments alongside the definition explaining why only this one algo is supported.

When we clean-up to use C-constant strings, the replacement should be a pass-thru.

Just a thought ... if this code consolidation is possible to be done for the new code added with this change.